src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -118,17 +120,22 @@
     @Argument(doc = "paired mode evaluation of variants (combine mates, when possible)", fullName = "paired-mode")
     private static boolean pairedMode = true;
 
+    @Argument(doc = "write BAM of rejected reads", fullName = "write-rejected-reads")
+    private static boolean writeRejectedReads = false;
+
     @VisibleForTesting static Reference reference;
     @VisibleForTesting static CodonTracker codonTracker; // device for turning SNVs into CodonVariations
+    @VisibleForTesting static SAMFileGATKReadWriter bamWriter;
 
     // a map of SNV sets onto number of observations of that set of variations
     private static final HopscotchMap<SNVCollectionCount, Long, SNVCollectionCount> variationCounts =
             new HopscotchMap<>(10000000);
 
-    private static final ReadCounts readCounts = new ReadCounts();
-    private static final MoleculeCounts moleculeCountsUnpaired = new MoleculeCounts();
-    private static final MoleculeCounts moleculeCountsDisjointPair = new MoleculeCounts();
-    private static final MoleculeCounts moleculeCountsOverlappingPair = new MoleculeCounts();
+    private static long totalBaseCalls;
+    private static final ReportTypeCounts readCounts = new ReportTypeCounts();
+    private static final ReportTypeCounts unpairedCounts = new ReportTypeCounts();
+    private static final ReportTypeCounts disjointPairCounts = new ReportTypeCounts();
```

Comment:
Is `disjoint` the same as `non-overlapping`, in the strict technical sense (i.e. hard clipping not included, soft clipping included)?

Reply:
Unclear on what you're asking:  We're only dealing with primary lines (no hard clips).  But we're also doing our own quality trimming and short-fragment trimming, so it's not just the cigar that dictates whether the evaluated bases in a pair overlap, abut, or are disjoint.

Reply:
I understand now what `disjoint` means. Thanks!